packages/styled-components/src/sheet/GroupedTag.js

```diff
@@ -0,0 +1,96 @@
+// @flow
+
+import type { Tag } from './Tag';
+
+/** Group-aware Tag that sorts rules by indices */
+export interface GroupedTag {
+  constructor(tag: Tag): void;
+  insertRules(group: number, rules: string[]): void;
+  clearGroup(group: number): void;
+  getGroup(group: number): string;
+  length: number;
+}
+
+/** Create a GroupedTag with an underlying Tag implementation */
+export const makeGroupedTag = (tag: Tag): GroupedTag => {
+  return new DefaultGroupedTag(tag);
+};
+
+const BASE_SIZE = 1 << 8;
+
+class DefaultGroupedTag implements GroupedTag {
+  groupSizes: Uint32Array;
+
+  length: number;
+
+  tag: Tag;
+
+  constructor(tag: Tag) {
+    this.groupSizes = new Uint32Array(BASE_SIZE);
```

Comment:
Did not realize IE10/11 supported this, pretty cool

Reply:
Yes! Super happy that they do :heart: I'll still test this on IE11 though just to be sure.